rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -160,7 +160,11 @@ void RSocketStateMachine::disconnect(folly::exception_wrapper ex) {
   }
 
   closeFrameTransport(std::move(ex), StreamCompletionSignal::CONNECTION_END);
-  pauseStreams();
+
+  if (connectionEvents_) {
+    connectionEvents_->onStreamsPaused(ex);
```

Comment:
parameter ex is already moved on line 162 so you are never sending an exception. I would suggest to remove the param.